src/plugins/Polls/UI/PollsInPost.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from 'react'
+import MaskbookPluginWrapper from '../../MaskbookPluginWrapper'
+import type { TypedMessage } from '../../../protocols/typed-message'
+import { renderWithPollMetadata, PollMetadataReader } from '../utils'
+import Services from '../../../extension/service'
+import type { PollGunDB } from '../Services'
+import { PollCardUI } from './Polls'
+import { PollMetaData, PollStatus } from '../types'
+
+interface PollsInPostProps {
+    message: TypedMessage
+}
+
+export default function PollsInPost(props: PollsInPostProps) {
+    const { message } = props
+    const [status, setStatus] = useState<PollStatus>(PollStatus.Inactive)
+    const [updatedPoll, setUpdatedPoll] = useState<PollMetaData | undefined>(undefined)
+
+    const vote = (poll: PollGunDB, index: number) => {
+        if (new Date().getTime() <= poll.end_time) {
+            setStatus(PollStatus.Voting)
+            Services.Plugin.invokePlugin('maskbook.polls', 'vote', {
+                poll,
+                index,
+            }).then((res) => {
+                setStatus(PollStatus.Voted)
+                setUpdatedPoll(res as PollMetaData)
+            })
+        } else {
+            setStatus(PollStatus.Closed)
+        }
+    }
+
+    useEffect(() => {
+        const { key } = PollMetadataReader(props.message.meta).val as PollMetaData
```

Comment:
Please don't do this. Either use `unwrap` or handle the error by `if (result.err)`